graviti/utility/binary_file.py

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
+#
+
+"""Graviti file related class."""
+
+import mimetypes
+from hashlib import sha1
+from pathlib import Path
+from typing import TYPE_CHECKING, Dict, Type, TypeVar, Union
+from urllib.request import url2pathname
+
+import pyarrow as pa
+from _io import BufferedReader
+
+from graviti.utility.repr import ReprMixin
+from graviti.utility.requests import UserResponse
+
+if TYPE_CHECKING:
+    from graviti.manager import Dataset
+
+_T = TypeVar("_T", bound="File")
+
+_ENCODINGS = mimetypes.encodings_map
+
+
+class FileBase(ReprMixin):
+    """This class represents the file in a DataFrame."""
+
+    _key: str
+    _format: str
+    _size: int
+
+    def open(self) -> Union[UserResponse, BufferedReader]:
+        """Return the binary file pointer of this file.
+
+        Raises:
+            NotImplementedError: The method of the base class should not be called.
+
+        """
+        raise NotImplementedError
+
+    @property
+    def key(self) -> str:
+        """Get the key of the file.
+
+        Returns:
+            The key of the file.
+
+        """
+        return self._key
+
+    @property
+    def format(self) -> str:
+        """Get the format of the file.
+
+        Returns:
+            The format of the file.
+
+        """
+        return self._format
+
+    @property
+    def size(self) -> int:
+        """Get the size of the file.
+
+        Returns:
+            The size of the file.
+
+        """
+        return self._size
+
+    @staticmethod
+    def from_pyarrow(scalar: pa.StructScalar, dataset: "Dataset") -> Union["RemoteFile", "File"]:
+        """Create File or RemoteFile instance with pyarrow struct scalar.
+
+        Arguments:
+            scalar: The input pyarrow struct scalar.
+            dataset: The dataset to which the file belongs.
+
+        Returns:
+            The File or RemoteFile instance.
+
+        """
+        key = scalar["key"].as_py()
+        format_: str = scalar["format"].as_py()
+        size: int = scalar["size"].as_py()
+
+        if key.startswith("file://"):
```

Comment:
the key of remote file is not starts with `oss://` or `s3://`, is it a good choice to add `file://` suffix for local file?

Reply:
But we need to distinguish the local file and the remote file with the key.

Reply:
we need a better way to distinguish the local and remote file.